src/Controller/Controller.hpp

```diff
@@ -17,25 +24,24 @@ class Controller
     EditMode
   };
 
-  Controller(JsonDocumentObserverPtr designDocObsever = JsonDocumentObserverPtr(),
-             JsonDocumentObserverPtr layoutDocObsever = JsonDocumentObserverPtr(),
-             RunMode mode = NormalMode);
+  Controller(std::shared_ptr<RunLoop> runLoop, Presenter& presenter, RunMode mode = NormalMode);
   ~Controller() = default;
 
   bool start(const std::string& filePath, const char* designDocSchemaFilePath = nullptr);
   bool start(const std::vector<char>& buffer, const char* designDocSchemaFilePath = nullptr);
 
-  void onClick(const std::string& path);
-
 private:
-  JsonDocumentObserverPtr m_design_doc_observer;
-  JsonDocumentObserverPtr m_layout_doc_observer;
+  std::shared_ptr<RunLoop> m_run_loop;
+  rxcpp::observer<ModelEventPtr> m_design_doc_observer;
```

Comment:
These observers should be reference of those of Presenter. The reference relationship should be made more clear.

Reply:
fixed